packages/vue/src/components/SpModal/SpModal.vue

```diff
@@ -0,0 +1,165 @@
+<template>
+  <transition name="modal-fade">
+    <div
+      v-if="visible"
+      class="modal-backdrop"
+      @click="$emit('close')"
+    >
+      <div
+        class="modal sp-box sp-shadow"
+        role="dialog"
+        @click.stop
+      >
+        <header
+          class="modal-header"
+          id="modalTitle"
+        >
+          <slot name="header">
+            <h3>{{ title || 'Default title' }}</h3>
+          </slot>
+          <SpLinkIcon
+            v-if="closeIcon"
+            icon="Close"
+            aria-label="Close modal"
+            class="close-icon"
+            @click="$emit('close')"
+          />
+        </header>
+
+        <section
+          class="modal-body"
+          id="modalDescription"
+        >
+          <slot name="body">
+            Default body
+          </slot>
+        </section>
+
+        <footer class="modal-footer">
+          <slot name="footer">
+            <SpButton
+              v-if="submitButton"
+              aria-label="Submit"
+              type="primary"
+              @click="$emit('submit')"
+            >
+              Submit
+            </SpButton>
+            <SpButton
+              v-if="cancelButton"
+              aria-label="Close modal"
+              type="secondary"
+              @click="$emit('close')"
+            >
+              Close
+            </SpButton>
+          </slot>
+        </footer>
+      </div>
+    </div>
+  </transition>
+</template>
+
+<script>
+export default {
+  name: 'SpModal',
+  props: {
+    visible: {
+      type: Boolean,
+      defaultsTo: false
+    },
+    title: {
+      type: String,
+      defaultsTo: null
+    },
+    submitButton: {
+      type: Boolean,
+      defaultsTo: true
+    },
+    closeIcon: {
+      type: Boolean,
+      defaultsTo: true
+    },
+    cancelButton: {
+      type: Boolean,
+      defaultsTo: true
+    },
+  },
+  data() {
+    return {
```

Comment:
we should add the emitting of `close` on `esc` press as well

Reply:
Updated